setuptools/config/_apply_pyprojecttoml.py

```diff
@@ -181,10 +182,12 @@ def _long_description(
         dist._referenced_files.add(file)
 
 
-def _license(dist: Distribution, val: dict, root_dir: StrPath | None):
+def _license(dist: Distribution, val: str | dict, root_dir: StrPath | None):
     from setuptools.config import expand
 
-    if "file" in val:
+    if isinstance(val, str):
+        _set_config(dist, "license_expression", _static.Str(val))
+    elif "file" in val:
```

Comment:
@cdce8p I updated this part of the code to solve the merge conflict. I hope that is OK.

Reply:
Thanks. Feel free to ping me, especially if one of my PRs need attention. I'm usually fairly quick to respond.

_Small side note. If I review PRs from other people, I only update them with merge commits. The rebase one makes it unnecessarily difficult for the OP to follow and understand what's changed on their own PR. I myself only do force pushes if my PR hasn't been reviewed yet, afterwards only merge commits._

If that's ok with you, I'll force push the rebase once #4796 is done. Just so I'm aware what has changed.

Reply:
> If that's ok with you, I'll force push the rebase once https://github.com/pypa/setuptools/pull/4796 is done. Just so I'm aware what has changed.

Thanks @cdce8p. That would be helpful.
I usually try to avoid merging the full main branch back to a PR, because it messes up the graph visualisation (which becomes difficult to follow), that is why I went with a rebase, sorry if that disrupted your workflow.

Reply:
> I usually try to avoid merging the full main branch back to a PR, because it messes up the graph visualisation (which becomes difficult to follow), that is why I went with a rebase, sorry if that disrupted your workflow.

Don't worry about it. Guess everyone has their own preferred workflow. Just wrote that as it took me a bit to actually see what you've changed during the rebase.

> Thanks @cdce8p. That would be helpful.

👍🏻